python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -216,19 +219,45 @@ def get(
         **kwargs: Any,
     ) -> Any:
         # Docstring inherited.
-        if not isinstance(datasetRefOrType, DatasetRef):
-            raise NotImplementedError("RemoteButler currently only supports get() of a DatasetRef")
-
-        dataset_id = datasetRefOrType.id
-        request = GetFileRequestModel(dataset_id=dataset_id)
-        response = self._post("get_file", request)
-        if response.status_code == 404:
-            raise LookupError(f"Dataset not found with ID ${dataset_id}")
+        if isinstance(datasetRefOrType, DatasetRef):
+            dataset_id = datasetRefOrType.id
+            response = self._get(f"get_file/{dataset_id}")
+            if response.status_code == 404:
+                raise LookupError(f"Dataset not found with ID {dataset_id}")
+        else:
+            request = GetFileByDataIdRequestModel(
+                dataset_type_name=self._normalize_dataset_type_name(datasetRefOrType),
+                collections=self._normalize_collections(collections),
+                data_id=self._simplify_dataId(dataId, kwargs),
+            )
+            response = self._post("get_file_by_data_id", request)
+            if response.status_code == 404:
+                raise LookupError(f"Dataset not found with DataId {dataId}")
+
         response.raise_for_status()
         model = self._parse_model(response, GetFileResponseModel)
 
+        # If the caller provided a DatasetRef or DatasetType, they may have
+        # overridden the storage class on it.  We need to respect this, if they
+        # haven't asked to re-override it.
+        explicitDatasetType = _extract_dataset_type(datasetRefOrType)
```

Comment:
We can avoid one of the isinstance checks if we retrieve the dataset type from the dataset ref we knew we got in the first branch.